### PR TITLE
Feature: link to previous blames in view blame page

### PR DESCRIPTION
**Pull request by rogerluo410**
Feature: 

1. Fix bottom lines mismatched between commit info and code line  in view blame page.
2. Add previous blame links

 Related PR:
[](url)https://github.com/go-gitea/gitea/pull/15148

Feature request issue:
[](url)https://github.com/go-gitea/gitea/issues/15109

Hi, @noerw,  I noticed you created a PR for the request, perhaps this one could more helpful to what we ask for.

Performance:
blame

### CODE REVIEW

**Comment by jolheiser** [comment]
Perhaps the [versions](https://octicons-primer.vercel.app/octicons/versions-16) octicon, to match GH?

**Comment by noerw** [comment]
The linter complains:
`routers/repo/blame.go:28:6: exported type BlameRow should have comment or be unexported`

**Comment by rogerluo410** [comment]
> The linter complains:
> `routers/repo/blame.go:28:6: exported type BlameRow should have comment or be unexported`

Fixed

**Comment by rogerluo410** [comment]
> Perhaps the [versions](https://octicons-primer.vercel.app/octicons/versions-16) octicon, to match GH?

Updated svg

**Comment by zeripath** [comment]
Please resolve conflicts.

File: web_src/less/_base.less [hunk 1]
```diff
@@ -1382,6 +1382,14 @@ a.ui.label:hover {
   margin-right: 0;
 }
 
+.lines-blame-btn {
+  padding-left: 10px;
+  padding-right: 10px;
+  text-align: right !important;
+  background-color: #f5f5f5;
```
Comment (noerw) [inline web_src/less/_base.less:1389]: This background color needs to overridden in `web_src/less/themes/them-arc-green.less` L464:
```
.blame .lines-num,
.blame .lines-blame-button {
```

File: routers/repo/blame.go [hunk 2]
```diff
@@ -218,33 +245,27 @@ func renderBlame(ctx *context.Context, blameParts []git.BlamePart, commitNames m
...
+
+			br.Code = gotemplate.HTML(line)
+			rows = append(rows, br)
 		}
 	}
 
-	ctx.Data["BlameContent"] = gotemplate.HTML(codeLines.String())
-	ctx.Data["BlameCommitInfo"] = gotemplate.HTML(commitInfo.String())
-	ctx.Data["BlameLineNums"] = gotemplate.HTML(lineNumbers.String())
+	ctx.Data["Codes"] = rows
```
Comment (noerw) [inline routers/repo/blame.go:269]: Can we rename this to `"Rows"`?

**Review by noerw — CHANGES_REQUESTED** [review]
One nit, otherwise looks good :)
Please merge current `main` branch

File: web_src/less/_base.less (hunk 1 above)
Comment (6543) [reply web_src/less/_base.less:1389]: i think we should use css vars

Comment (6543) [reply web_src/less/_base.less:1389]: silverwind did the transition to them for colors ...

So you dont have to touch our dark theme!

Comment (noerw) [reply web_src/less/_base.less:1389]: In that case, the existing override for `.blame .lines-num` in `theme-arc-green.less` can probably dropped

**Comment by 6543** [comment]
@noerw @rogerluo410 thanks for the work -> merged as 9c6aeb4

**Pull request closed without merging**